sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -487,38 +487,14 @@ private boolean canProduceMultiplePanes(WindowingStrategy<?, ?> strategy) {
     public void populateDisplayData(DisplayData.Builder builder) {
       super.populateDisplayData(builder);
 
-      if (windowFn != null) {
-        builder
-            .add(DisplayData.item("windowFn", windowFn.getClass())
-              .withLabel("Windowing Function"))
-            .include("windowFn", windowFn);
-      }
-
-      if (allowedLateness != null) {
-        builder.addIfNotDefault(DisplayData.item("allowedLateness", allowedLateness)
-              .withLabel("Allowed Lateness"),
-            Duration.millis(BoundedWindow.TIMESTAMP_MAX_VALUE.getMillis()));
-      }
-
-      if (trigger != null && !(trigger instanceof DefaultTrigger)) {
-        builder.add(DisplayData.item("trigger", trigger.toString())
-          .withLabel("Trigger"));
-      }
-
-      if (mode != null) {
-        builder.add(DisplayData.item("accumulationMode", mode.toString())
-          .withLabel("Accumulation Mode"));
-      }
-
-      if (closingBehavior != null) {
-        builder.add(DisplayData.item("closingBehavior", closingBehavior.toString())
-          .withLabel("Window Closing Behavior"));
-      }
-
-      if (outputTimeFn != null) {
-        builder.add(DisplayData.item("outputTimeFn", outputTimeFn.getClass())
-          .withLabel("Output Time Function"));
-      }
+      populateWindowingStrategyDisplayData(
+          WindowingStrategy.of(windowFn)
+              .withAllowedLateness(allowedLateness)
```

Comment:
We do lack the context here to construct the entire windowing strategy, which is unfortunate. `Window.Assign` is provided it on construction, but I don't know if there's any way to pass it to the identity flatten's display data.